.travis.yml

```diff
@@ -8,29 +7,114 @@ services:
+    # AWS_SECRET_ACCESS_KEY
+    - secure: "gYcMV5o/iI3PZ09RLop9FBS91z+Iy3hdgT930ZTWuYz8QKt8vV5eKW1PSRRvVRmHkiBGdrKYmXfiJGa+2MGMp9v4sHhWkPTSyg5279f+/piT4ccoXT/PyabYB8sHQ4SRQThUKDaVOMy3oPUB43IaebEzPL2GMgbSMfOLj6v5MuhkMBslVg1/pVUXxiZbH1bIB4S8ZtaM7iWY0Ornm9kAb4OwhqifkVGh99p76+MwKT139LFrsXzM5AoZSLucRlVwnSlckpZy+jXVD1HRr6LCyneiWqSpdmSUXmLDP0XcBtfd7aoabawlaB3AUJRpgoAZUy4aVd9FPwnpYglvpnKPK37hx0oy43bS6y4k7NI/tdhYfyN0L4yGC4EYiOTkMhK9ja083J1z17P7zNr0jxTUIzpvptROc5oUVfznJt+VpzCdMagodzNqmgeg6rytjkVRscc2OJH78nVSAIRMteN7E9V6bTVsKpe7BR4eFPnU/4n76lZ/IzPZXAfIzK7vu5dVC8A1WvhpBAApUB7JJay+keBGRqMDd4Fn5o0BSUQoqPuyw6G/cIvJqzX2iZC98KUjVTW/5WI0YrLxYnFNuUKYF+npoSq/KHbTPJQj4SRYFmi1rCfA39PYBeVqksTNL4ouTfgAxwiB3z4RCaYi++oRnjqRReY1u433xzotGDJ6RUE="
 
 before_install:
   - sudo sysctl -w vm.max_map_count=262144
+  - ./.ci/start_kuzzle.sh
+
+addons:
+  apt:
+    packages:
+    - python
+    - python-pip
+
+install:
+  - pip install awscli --upgrade --user
+
+before_deploy:
+  - docker --run -it -v "$(pwd)":/mnt kuzzleio/sdk-cross:gcc make package
 
 deploy:
   provider: script
   script: cd build/java && sudo gradle bintrayUpload
   skip_cleanup: true
+  on:
+    condition: "$JDK = openjdk8"
+
+  provider: s3
+  access_key_id: $AWS_ACCESS_KEY_ID
+  secret_access_key: $AWS_SECRET_ACCESS_KEY
+  bucket: $AWS_S3_BUCKET
+  region: us-west-2
+  skip_cleanup: true
+  upload-dir: sdk/java/$TRAVIS_BRANCH
+  local-dir: deploy
+  on:
+    condition: "$JDK = openjdk8"
+
+after_deploy:
+  - aws cloudfront create-invalidation --distribution-id $AWS_CLOUDFRONT_DISTRIBUTION_ID --paths "/*"
 
 matrix:
   include:
+# ---------------------------------------------
+# OpenJDK 8: Build, test and deploy (only on master)
+# ---------------------------------------------
+# AMD64
+# ---------------------------------------------
+    - env:
+        - ARCH="amd64"
+        - JDK="openjdk8"
+      script:
+        - docker run --rm -it --network ci_default --link kuzzle -v "$(pwd)":/mnt kuzzleio/sdk-cross:openjdk8 bash -c "make clean all && make test"
+
+# ---------------------------------------------
+# I386
+# ---------------------------------------------
+    - env:
+        - ARCH="x86"
+        - JDK="openjdk8"
+      script:
+        - docker run --rm -it --network ci_default --link kuzzle -v "$(pwd)":/mnt kuzzleio/sdk-cross:openjdk8-i386 bash -c "make clean all && make test"
```

Comment:
we should rename our docker image tags, as `i386` is not a suitable name for x86 architecture